filclient.go

```diff
@@ -219,33 +219,22 @@ func (fc *FilClient) streamToMiner(ctx context.Context, maddr address.Address, p
 	return s, nil
 }
 
-type ApiError struct {
-	Under error
-}
-
-func (e *ApiError) Error() string {
-	return fmt.Sprintf("api error: %s", e.Under)
-}
-
-func (e *ApiError) Unwrap() error {
-	return e.Under
-}
-
+// Errors - ErrMinerConnectionFailed, ErrLotusError
 func (fc *FilClient) connectToMiner(ctx context.Context, maddr address.Address) (peer.ID, error) {
 	minfo, err := fc.api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
 	if err != nil {
-		return "", &ApiError{err}
+		return "", NewErrMinerConnectionFailed(NewErrLotusError(err))
```

Comment:
I think in this case, this can be considered an api failure, which we wouldnt necessarily hold against the miner.